### ISSUE TITLE
package_hooks_linter should trigger on .onUnload without arguments

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2940]
```r
lint(text = ".onUnload <- function() NULL", linters = package_hooks_linter())
# ℹ No lints found.
```

But such a hook triggers an `R CMD check` `WARNING`

```
* checking whether the namespace can be unloaded cleanly ... WARNING
---- unloading
Warning message:
.onUnload failed in unloadNamespace() for '${PKG}', details:
  call: fun(...)
  error: unused argument ("/tmp/Rtmp1ax8VR/${PKG}.Rcheck/${PKG}") 
```
